linker/Mono.Linker.Steps/MarkStep.cs

```diff
@@ -115,6 +115,22 @@ void Process ()
 				ProcessQueue ();
 				ProcessVirtualMethods ();
 			}
+
+			// deal with [TypeForwardedTo] pseudo-attributes
+			foreach (AssemblyDefinition assembly in _context.GetAssemblies ()) {
+				if (!assembly.MainModule.HasExportedTypes)
+					continue;
+
+				foreach (var exported in assembly.MainModule.ExportedTypes) {
+					if (!exported.IsForwarder)
+						continue;
+					var type = exported.Resolve ();
+					if (!Annotations.IsMarked (type))
+						continue;
+					Annotations.Mark (exported);
+					Annotations.Mark (assembly.MainModule);
```

Comment:
I looked some more into code that deals with type forwarders. I believe the intention is that if nothing in an assembly is marked except type forwarders, then all references to the type forwarders in other assemblies should be resolved in SweepStep.SweepReferences and the assembly can be deleted. There is another piece of code in SweepStep.Process that resolves all references in Copy assemblies to make sure they don't have references to type forwarders in deleted assemblies.

I have some scenarios where I don't want this optimization to kick in and need to keep assemblies with type forwarders and not resolve references to them. So, I'll create another pull request where I will add a command line switch to disable this optimization. assembly.MainModule above will only be marked when the switch is passed. Likewise, the code resolving references in  SweepStep.Process and SweepStep.SweepReferences will only run if the new switch is not passed.